txcache/wrappedTransaction.go

```diff
@@ -18,19 +19,29 @@ type WrappedTransaction struct {
 	ReceiverShardID uint32
 	Size            int64
 
+	Fee          atomic.Pointer[big.Int]
 	PricePerUnit atomic.Uint64
+	IsGuarded    bool
 }
 
 // precomputeFields computes (and caches) the (average) price per gas unit.
 func (wrappedTx *WrappedTransaction) precomputeFields(txGasHandler TxGasHandler) {
-	fee := txGasHandler.ComputeTxFee(wrappedTx.Tx).Uint64()
+	fee := txGasHandler.ComputeTxFee(wrappedTx.Tx)
 
 	gasLimit := wrappedTx.Tx.GetGasLimit()
 	if gasLimit == 0 {
 		return
 	}
 
-	wrappedTx.PricePerUnit.Store(fee / gasLimit)
+	wrappedTx.Fee.Store(fee)
+	wrappedTx.PricePerUnit.Store(fee.Uint64() / gasLimit)
+
+	txAsGuardedTransaction, ok := wrappedTx.Tx.(data.GuardedTransactionHandler)
```

Comment:
Temporary workaround. Will be dropped in the future (upon adjusting the interceptors).